flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/serde/ContextResolvedTableSerdeTest.java

```diff
@@ -191,6 +176,24 @@ void withAnonymousTable() {
                             TableConfigOptions.CatalogPlanCompilation.IDENTIFIER,
                             TableConfigOptions.CatalogPlanRestore.IDENTIFIER);
 
+            @Test
+            void withAnonymousTable() throws Exception {
+                byte[] actualSerialized =
+                        createObjectWriter(ctx).writeValueAsBytes(ANONYMOUS_CONTEXT_RESOLVED_TABLE);
+
+                assertThatThrownBy(
+                                () ->
+                                        createObjectReader(ctx)
+                                                .readValue(
+                                                        actualSerialized,
+                                                        ContextResolvedTable.class))
+                        .satisfies(
+                                anyCauseMatches(
+                                        ValidationException.class,
+                                        ContextResolvedTableJsonDeserializer.missingIdentifier()
```

Comment:
maybe there is still some misunderstanding, but why do we throw an exception here? anonymous tables have nothing to do with the test matrix for catalog objects, the given option should have no effect.